### PR TITLE
Bylaws: revise indemnification language

### PR DESCRIPTION
**Pull request by ebullient**
[🗳️ Vote progress](https://github.com/commonhaus/foundation/pull/187#issuecomment-2273807144 "IC_kwDOKRPTI86Hh4so")

For some context, I was thinking about our language after reading this: 

https://pyfound.blogspot.com/2024/07/notice-of-python-software-foundation.html
  
We were already reasonably constrained, but it seemed like constraining required/mandatory indemnification to the council/directors and officers at first makes the most sense. We can then allow the CFC to extend that indemnification to cover others (employees or committee members) when appropriate without requiring revisions to bylaws.

The revised text also clarifies that advance payments for indemnifications must be made in writing and include an acknowledgement to repay those funds if necessary

The language brings in more legal jargon, but I've included TIPs to try and keep things readable.

voting group: @commonhaus/cf-egc 

Do one of the following:

- Approve the PR or react with 👍 (:+1:) if it looks good to you
- Review with Comments or react with 👀 (:eyes:) if you're "ok" with it (it may not be your favorite)
- If you think it needs discussion or revision
    - Create a review, add your comments and require changes
    - Use the +- button to make a suggestion (instead of just adding a comment).

### CODE REVIEW

**Review by evanchooly — APPROVED** [review]

**Review by kenfinnigan — APPROVED** [review]

File: bylaws/7-indemnification-dissolution.md [hunk 1]
```diff
@@ -22,54 +22,69 @@ The CF Council (CFC) shall assess the Foundation's risks and evaluate whether to
...
-(2) had no reason to believe their conduct was unlawful, in the case of criminal matters.
+
+1. acted in good faith and in a manner they reasonably believed to be consistent with the best interests of the Foundation, and
+2. had no reason to believe their conduct was unlawful, in the case of criminal matters.
 
 This indemnification does not extend to a person who is found liable to the Foundation in an action brought by the Foundation, or to enforce the Foundation's rights.
 Otherwise, the outcome of an action will not alone create a presumption that a person did not act in good faith, in the best interests of the Foundation, or with lawful intent.
 
+> [!TIP]
+> If the Foundation sues someone or they are found liable to the Foundation, they won't be covered. Just being sued doesn't mean they did something wrong.
```
Comment (henri-tremblay) [inline bylaws/7-indemnification-dissolution.md:50]: Who is "they" in `they are found liable to the Foundation`? Is "sued" the official term to use?

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md:50]: The "TIP" uses less formal/colloquial language. It is an attempt to help folks orient so they can understand what they are reading.

Comment (maxandersen) [inline bylaws/7-indemnification-dissolution.md:50]:
```suggestion
> If the Foundation has a litigation with someone or someone are found liable to the Foundation, that someone won't be covered. Important to remember that just participating in a litigation does not mean you are guilty.
```

i pondered on this phrasing too as "they" is ambigous; especially in the latter statement ("they" can be both foundation or someone else). Above is an attempt to make it more clear. Still not fully happy but it is less ambigious.

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md:50]: I see where you're going with this. Will accept and have another go.

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md]: @henri-tremblay and @maxandersen .. pushed revised language. Better?

Comment (henri-tremblay) [reply bylaws/7-indemnification-dissolution.md]: `someone is found liable` no? Apart from that, it's clearer.

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md]: This is the new language: 

> [!TIP]
> If the Foundation brings a lawsuit against a person, or if the person is found responsible for wrongdoing in such a lawsuit, they will not be covered by this indemnification.
> However, being involved in a lawsuit does not mean the person did something wrong.

**Review by cealsair — APPROVED** [review]

File: bylaws/7-indemnification-dissolution.md (hunk 1 above)
Comment (henri-tremblay) [reply bylaws/7-indemnification-dissolution.md]: Better. I'm still unhappy with the `However` part but the first sentence is perfect.

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md]: I was on the fence about that "However". I'll remove it.

File: bylaws/7-indemnification-dissolution.md [hunk 2]
```diff
@@ -22,54 +22,71 @@ The CF Council (CFC) shall assess the Foundation's risks and evaluate whether to
...
+This indemnification extends to all reasonable expenses (like legal fees), judgments, fines, and settlement amounts that a person may have to pay in connection with a legal action, so long as the person:
+
+1. acted in good faith and in a manner they reasonably believed to be consistent with the best interests of the Foundation, and
+2. had no reason to believe their conduct was unlawful, in the case of criminal matters.
 
 This indemnification does not extend to a person who is found liable to the Foundation in an action brought by the Foundation, or to enforce the Foundation's rights.
 Otherwise, the outcome of an action will not alone create a presumption that a person did not act in good faith, in the best interests of the Foundation, or with lawful intent.
 
+> [!TIP]
+> If the Foundation brings a lawsuit against a person, or if the person is found responsible for wrongdoing in such a lawsuit, they will not be covered by this indemnification.
```
Comment (henri-tremblay) [inline bylaws/7-indemnification-dissolution.md:51]: Hum, we are back to "they". They is the person? Also, do it applies to legal entities or only to physical entities?

Comment (henri-tremblay) [inline bylaws/7-indemnification-dissolution.md:51]:
```suggestion
> If the Foundation brings a lawsuit against a person, or if the person is found responsible for wrongdoing in such a lawsuit, that person will not be covered by this indemnification.
```

Comment (ebullient) [reply bylaws/7-indemnification-dissolution.md]: Commonhaus is indemnifying people (as an organization, our members and representatives are people), not companies or other legal entities.

**Comment by gavinking** [comment]
At least naively, this seems insufficient when it comes to possible patent litigation.

**Comment by gavinking** [comment]
There's also quite a large asymmetry here: CF requires that *everyone* abides by all its various policies.

But when it comes to indemnification, only certain special people are presumed indemnified.

**Comment by gavinking** [comment]
How about: every single person who is subject to the requirement to comply with CF policies is a priori assumed indemnified by the foundation when carrying out foundation-related activities according to those policies?

**Comment by ebullient** [comment]
"Every person that has to abide by the policies" or even "Every member of the foundation" is too broad of a group. Some other groups have recently realized they were also over-broad, which is what made me think of this: https://pyfound.blogspot.com/2024/07/notice-of-python-software-foundation.html

We're trying to find a balance here. The Foundation must indemnify members of the council (who are the most responsible for ensuring the organization runs properly). This language does allow us to extend that indemnification to people (committee members, staff, etc) who are acting on behalf of the foundation.

We can not be required, by the bylaws, to indemnify everyone.

**Comment by gavinking** [comment]
> I would pursue a different route entirely (and would add that to the IP Policy).

Well OK but this has to be dealt with *somewhere* and currently it's not.

If a community member gets sued by a patent troll for their contribution to a CF project, I don't think it's unreasonable to expect the foundation to step up and help defend them.

**Comment by ebullient** [comment]
I think that is an unreasonable expectation, actually. 

We can not promise to (or be required to) indemnify every contributor to any project from action by patent trolls. That can put the foundation (as a whole) at risk.

**Comment by gavinking** [comment]
Well I'm struggling to see how that's not a real problem.

If Red Hat donates Quarkus to CF, then suddenly Red Hat employees become potentially unprotected in patent litigation? (Previously they would have been protected due to their role as RH employees.)

Tell me what's wrong with my reasoning here.

**Comment by gavinking** [comment]
I mean: I suppose they're still protected by RH if their contributions are considered a part of their work at RH. Not sure how any of this works.

**Comment by ebullient** [comment]
While IANAL, I believe they would still be protected due to their role as RH employees (as you stated). After that, my thought is that Commonhaus should join the Open Invention Network, and gain some protection that way.

- https://openinventionnetwork.com/
- https://www.zdnet.com/article/open-invention-network-comes-to-gnomes-aid-in-patent-troll-fight/

File: bylaws/8-indemnification-dissolution.md [hunk 3]
```diff
@@ -22,54 +22,74 @@ The CF Council (CFC) shall assess the Foundation's risks and evaluate whether to
...
+> The council can choose to extend this coverage to staff and [committee members][committee] if they decide it’s appropriate.
+
+[officers]: ./3-cf-council.md#officer-roles-and-appointment
+[committee]: ./3-cf-council.md#committees-and-delegation
 
 **Standard of Conduct.**
-This indemnification extends to all reasonable expenses (including attorneys’ fees), judgments, fines, and other amounts paid in settlement by the person in connection with the action, so long as the person:
-(1) acted in good faith and in a manner they reasonably believed to be consistent with the best interests of the Foundation, and
-(2) had no reason to believe their conduct was unlawful, in the case of criminal matters.
+
```
Comment (cealsair) [inline bylaws/8-indemnification-dissolution.md:44]: In this section, should we mention that the person also needs to have followed the CF Code of Conduct and other related CF policies, such as Antitrust policy, Conflict of Interest, etc.?

Item 1 is very generic and doesn't mention any of the above.

**Review by cealsair — APPROVED** [review]

**Comment by ebullient** [comment]
vote::result

- We will raise another PR if Gavin has additional suggestions after consulting with SFC
- After discussion with @jkschneider, we can/should proceed on some items w/o waiting for @timtebeek

**Pull request merged**